### PR TITLE
[JENKINS-52205] replace Plugin with a GlobalConfiguration

### PR DESCRIPTION
**Pull request by ndeloof**
replace `extends Plugin` with `GlobalConfiguration` (as recommended by Plugin's javadoc)
which makes locale plugin magically JCasC-compliant :)

### CODE REVIEW

**Comment by ndeloof** [comment]
cc @oleg-nenashev

File: src/main/java/hudson/plugins/locale/PluginImpl.java [hunk 1]
```diff
@@ -20,7 +24,8 @@
 /**
  * @author Kohsuke Kawaguchi
  */
-public class PluginImpl extends Plugin {
+@Extension
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/locale/PluginImpl.java]: Add `@Symbol`?

Comment (ndeloof) [reply src/main/java/hudson/plugins/locale/PluginImpl.java]: require to upgrade core to 2.2+, is it ok ?

Comment (oleg-nenashev) [reply src/main/java/hudson/plugins/locale/PluginImpl.java]: Yes. I would just go straight to Java 8 and 2.60.3

**Comment by ndeloof** [comment]
@oleg-nenashev maybe I missed something but the Plugin implementation already used XStream persistence and migrating to a GlobalConfiguration won't change the xml file in use, so no data migration required.

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/locale-plugin/pull/7 does not work after the change, the `locale.xml` file just gets ignored. My proposal would be to keep the existing file and `PluginImlp` as is, but to migrate the configuration to a new class. I may be able to implement it in a compatible way

**Comment by ndeloof** [comment]
oh, Plugin.getConfigFile uses shortName for file name, while GlobalConfiguration relies on getId = FQCN. So we just need to override getConfigFile (or getId ?)

**Comment by ndeloof** [comment]
```
hudson.remoting.Channel$CallSiteStackTrace: Remote call to ubuntu-jenkinsinfra523432
NoClassDefFoundError: Could not initialize class com.sun.proxy.$Proxy8
```

I'm so much in love with Jenkins remoting ...

**Comment by bdominguez** [comment]
Any news on this?

**Comment by darxriggs** [comment]
@ndeloof & @oleg-nenashev ping

**Comment by ndeloof** [comment]
inactive, feel free to take over

**Pull request closed without merging**